lib/spack/spack/test/cmd_extensions.py

```diff
@@ -80,32 +95,102 @@ def hello(parser, args):
     elif args.subcommand == 'global':
         print(global_message)
 """)
-    implementation = extension_root.ensure('testcommand', 'implementation.py')
+    root.ensure('testcommand', '__init__.py')
+    implementation = root.ensure('testcommand', 'implementation.py')
     implementation.write("""
 def hello_world():
     print('Hello world!')
 
 def hello_folks():
     print('Hello folks!')
 """)
-    list_of_modules = list(sys.modules.keys())
-    with spack.config.override('config:extensions', [str(extension_root)]):
-        yield spack.main.SpackCommand('hello')
-
-    to_be_deleted = [x for x in sys.modules if x not in list_of_modules]
-    for module_name in to_be_deleted:
-        del sys.modules[module_name]
+    yield spack.main.SpackCommand('hello')
 
 
 def test_simple_command_extension(hello_world_cmd):
     output = hello_world_cmd()
     assert 'Hello world!' in output
 
 
+def test_multi_extension_search(hello_world_extension, tmpdir):
+    """Ensure we can find an extension command even if we search somewhere
+    it's isn't first.
+    """
+    extra_ext = tmpdir.mkdir('spack-testcommand2')
+    extra_ext.ensure('testcommand2', 'cmd', dir=True)
+    with spack.config.override('config:extensions',
+                               [str(extra_ext), str(hello_world_extension)]):
+        assert('Hello world') in spack.main.SpackCommand('hello-world')()
+
+
+def test_duplicate_module_load(hello_world_cmd):
+    """Ensure we correctly deal with duplicate command load attempts."""
```

Comment:
Can you update this docstring to rephrase in terms of what is correct (i.e. explain how it is correct without using the word "correct"):

> Ensure that the same command can be loaded multiple times without failure

to that end, it might make sense to invoke the command after loading it twice.